### ISSUE TITLE
Dependencies are not correctly linked to the incoming request

### ISSUE THREAD

**Issue by fgheysels** [arcus-azure/arcus.observability#387]
When logging dependency calls from within a Web API, I'd assume that the dependencies are correctly linked to the incoming request as being the parent of the dependency call.

Something like this:

![image](https://user-images.githubusercontent.com/3605786/172631474-06d99832-9ba8-46af-ba4a-5ed1aeb4a478.png)

What we see here, is that we have an incoming HTTP POST request, and the operation that is triggered by that request, performs 2 calls to a SQL Server database.
As you can see, those 2 SQL calls are indented as opposed to the incoming request.   This is because the ParentId of these 2 dependencies is set to the Id of the request.  (This is an example of built-in logging when using Entity Framework).
When looking at the details of such a SQL Dependency, we can see this:
![image](https://user-images.githubusercontent.com/3605786/172632018-88d8eb07-2848-4280-9924-9490796d91aa.png)

When looking at the details of the incoming request, we see this:
![image](https://user-images.githubusercontent.com/3605786/172632233-6905ed71-1580-49f3-8bb6-3b5f6494ae34.png)

However, when we use Arcus.Observability in a Web API project where we log a dependency ourselves from an API operation, like this:

```csharp
[HttpPost]
public IActionResult CreateCar()
{
    using (var m = DurationMeasurement.Start())
    {
        _logger.LogInformation(Activity.Current.ParentId);
        _logger.LogSqlDependency("test", "database", "something", "foo", true, m);
    }
    return Ok();
}
```

This gets logged like this:

![image](https://user-images.githubusercontent.com/3605786/172632726-241699d0-4f53-426f-9843-2024f85420d9.png)

The SQL Dependency here is not logged but has no parent reference to the request:
![image](https://user-images.githubusercontent.com/3605786/172632948-285dc1ae-135a-4e31-995b-93de9b01f595.png)

(Note: the fact that the traces are not in the correct order is mentioned here https://github.com/arcus-azure/arcus.webapi/issues/333 )

**Comment by fgheysels** [comment]
I've setup a simple Asp.Net core Web API project using the MIcrosoft template (so no Arcus is involved), and this works out of the box as expected when we use the `TelemetryClient` for tracking dependencies.

Relevant code:

```csharp
public static void Main(string[] args)
{
     var builder = WebApplication.CreateBuilder(args);

     // Add services to the container.
     builder.Services.AddApplicationInsightsTelemetry("9ab5198f-1529-473f-bd3f-9b5ef2d886c9");
     builder.Services.AddControllers();
   ....
}
```

```csharp
    [ApiController]
    [Route("[controller]")]
    public class WeatherForecastController : ControllerBase
    {       
        private TelemetryClient _logClient;

        public WeatherForecastController(TelemetryClient client)
        {
            _logClient = client;
        }
        
        [HttpGet(Name = "GetWeatherForecast")]
        public IActionResult Get()
        {
            _logClient.TrackDependency("SqlServer", "mysqlserver", "abc", DateTimeOffset.Now, TimeSpan.FromSeconds(2), true);

            return Ok();
        }
    }
```

Results in this in App Insights:

![image](https://user-images.githubusercontent.com/3605786/172696733-5652461e-3735-4f17-9cc3-039d152124cb.png)


![image](https://user-images.githubusercontent.com/3605786/172696626-aeda43ce-e5ca-4c49-b4ef-277dac79d2e4.png)

**Comment by stijnmoreels** [comment]
I'm wondering... we have our HTTP correlation that extracts the operation parent ID from the operation ID, so that the operation parent ID is available as a correlation model and will eventually lead to the telemetry setting of the operation parent ID. But that means that the initial operation ID would already contain a operation parent ID.

So, my question would be: did you provide an operation parent ID to the HTTP request?

**Comment by fgheysels** [comment]
These examples do not originate from the service-to-service POC.
First screenshot in the 1st post is an example of how logs ar written to AppInsights by Entity Framework.

The screenshots in the 2nd post are how it is done via Application Insights, without using Arcus.
Both samples show how it should be done, but when using Arcus.Observability, the parent is not set correctly and ideally, this should all happen under the hood.

**Comment by stijnmoreels** [comment]
Seems that this is an issue for the concrete implementations of observability, and not the observability itself, as it's about requests and how they extract correlation that can be used further to the tracking of dependencies.
So, Web API and Messaging, in this case.

**Comment by stijnmoreels** [comment]
Reproduced locally, seems indeed that this is something for the Web API repo, and not for this; as the Web API will have to generate something for the Observability functionality to use; same for the Messaging if there's no Operation Parent Id present. To be discussed further offline in a couple of hours 😉 .

**Comment by stijnmoreels** [comment]
Fixed with #411 , right?
